src/actions/disconnect.ts

```diff
@@ -76,66 +78,15 @@ export class DisconnectAction {
    */
   disconnectBlocks(workspace: WorkspaceSvg) {
     const cursor = workspace.getCursor();
-    if (!cursor) {
-      return;
-    }
-    let curNode: IFocusableNode | null = cursor.getCurNode();
-    let wasVisitingConnection = true;
-    while (
-      curNode &&
-      !(curNode instanceof RenderedConnection && curNode.isConnected())
-    ) {
-      if (curNode instanceof BlockSvg) {
-        const previous = curNode.previousConnection;
-        const output = curNode.outputConnection;
-        if (previous?.isConnected()) {
-          curNode = previous;
-          break;
-        } else if (output?.isConnected()) {
-          curNode = output;
-          break;
-        }
-      }
-
-      curNode = workspace.getNavigator().getParent(curNode);
-      wasVisitingConnection = false;
-    }
-    if (!curNode) {
-      console.log('Unable to find a connection to disconnect');
-      return;
-    }
-    if (!(curNode instanceof RenderedConnection && curNode.isConnected())) {
-      return;
-    }
-    const targetConnection = curNode.targetConnection;
-    if (!targetConnection) {
-      throw new Error('Must have target if connected');
-    }
-
-    const superiorConnection = curNode.isSuperior()
-      ? curNode
-      : targetConnection;
-
-    const inferiorConnection = curNode.isSuperior()
-      ? targetConnection
-      : curNode;
-
-    if (inferiorConnection.getSourceBlock().isShadow()) {
-      return;
-    }
-
-    if (!inferiorConnection.getSourceBlock().isMovable()) {
-      return;
-    }
-
-    superiorConnection.disconnect();
-    inferiorConnection.bumpAwayFrom(superiorConnection);
+    if (!cursor) return;
+    const curNode = cursor.getCurNode();
+    if (!(curNode instanceof BlockSvg)) return;
 
-    const rootBlock = superiorConnection.getSourceBlock().getRootBlock();
-    rootBlock.bringToFront();
+    const healStack = !curNode.outputConnection?.isConnected();
+    Events.setGroup(true);
```

Comment:
Set the event group to false afterwards.

Reply:
79a1c22db79b3daf26770913f02424b1d335c6f8